engine/src/main/java/org/terasology/engine/rendering/nui/layers/mainMenu/advancedGameSetupScreen/AdvancedGameSetupScreen.java

```diff
@@ -117,6 +120,19 @@ public void onOpened() {
         filterModules();
     }
 
+    @Override
+    public void onScreenOpened() {
+        super.onScreenOpened();
+
+        final UIText seed = find("seed", UIText.class);
+        if (seed != null) {
```

Comment:
Hm, if we need to check for `null` here, should we at least log a warning or something in case the `seed` text widget could not be found? I mean, it should be there when the screen is opened, right?

Reply:
tbh I'm not sure, why we need the null check(s) (there's one in AdvancedGameSetupScreen::initialise, too)... I would agree that the text widget should be there when the screen is opened and it should even already be there when it is initialized...
I don't see a valid reason why it should ever not be, so I'll remove the null checks.

Reply:
I think the main reason is that we don't have automated tests that spot NPEs like this. For instance, they can appear if we rename the UI control in the some `.ui` file but not here, or vice versa, move UI controls around but forget about the code, etc.

It will be pretty apparent while testing locally, but we don't have automated tests that "click through all the screens" that would catch that...